### DISCUSSION TITLE
How to achieve floating ui size and hide functionality in ark ui select, combobox.

### DISCUSSION THREAD

**Discussion by raja-OX** [chakra-ui/ark#2496] [General]
I want to floating ui functionality of size functionality fit to view port and hide functionality when reference is hidden. Is this possible in ark ui?

floating ui hide functionality.
```ts
const {refs, floatingStyles, middlewareData} = useFloating({
  middleware: [hide()],
});
 
return (
  <div
    ref={refs.setFloating}
    style={{
      ...floatingStyles,
      visibility: middlewareData.hide?.referenceHidden
        ? 'hidden'
        : 'visible',
    }}
  />
);
```

floating ui Size functionality.
```ts
const [maxHeight, setMaxHeight] = useState(null);
 
const {refs, floatingStyles} = useFloating({
  middleware: [
    size({
      apply({availableHeight}) {
        flushSync(() => setMaxHeight(availableHeight));
      },
    }),
  ],
});
 
return (
  <div
    ref={refs.setFloating}
    style={{...floatingStyles, maxHeight}}
  />
);
```

**Comment by segunadebayo** [comment]
We don't currently support the hide functionality (looking into it)

For the available height, please use the exposed css variable `var(--available-height)` as the value of `maxHeight` for the content.

**Comment by segunadebayo** [comment]
Closing — the `positioning` prop on Select supports `fitViewport`, `sameWidth`, and other Floating UI middleware options.
